### PR TITLE
Lookup AWS AccountNumber for the current user

### PR DESCRIPTION
**Pull request by EduardSergeev**
To be used for https://github.com/ambiata/slate/issues/42

### CODE REVIEW

**Comment by olorin** [comment]
LGTM. :shaved_ice: on :green_apple: (though will probably need to fix the permissions before the test passes).

**Comment by EduardSergeev** [comment]
> though will probably need to fix the permissions before the test passes

[ci seems to have necessary permissions](https://ci.ambiata.com/job/mismi-iam.branches/79/console)

**Comment by olorin** [comment]
So it does. :+1:

File: mismi-iam/src/Mismi/IAM/Data.hs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    ARN(..)
+  , AccountNumber(..)
+  ) where
+
+import           Mismi.Amazonka
+
+import           P
+
+
+newtype ARN = ARN {
```
Comment (charleso) [inline mismi-iam/src/Mismi/IAM/Data.hs]: Not used? Or you're adding this but it's unrelated?

Comment (EduardSergeev) [inline mismi-iam/src/Mismi/IAM/Data.hs]: Nope. Extra bit copied from slate. Will remove it from here, thanks!

**Comment by EduardSergeev** [comment]
Bump @charleso @nhibberd ok to merge?

**Comment by charleso** [comment]
I'm good with the code. There's a bigger question around what boxes can/should have this permission, and whether exposing this function will lead to deployment issues.

**Comment by EduardSergeev** [comment]
Hm... looks like it might not even work within the current environment: [Must specify userName when calling with non-User credentials](http://jenkins.ambiata.com/job/mismi-iam/7/console)

**Comment by nhibberd** [comment]
Previously would have been called with the user credentials, so that would make sense

**Pull request merged**